lib/resty/core/request.lua

```diff
@@ -346,6 +349,25 @@ function ngx.req.clear_header(name)
 end
 
 
+function ngx.req.get_uri_exten()
```

Comment:
We should no longer injecting new Lua API functions into the global `ngx.` table. It adds GC overhead to all the OpenResty Lua apps. We should put it into a separate Lua module which can be loaded only on demand. See the existing `ngx.re` module in the file `lib/ngx/re.lua` file. I guess it could be used this way:

```lua
local ngx_req = require "ngx.req"
...
say(ngx_req.get_uri_ext())
```